src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,502 @@
+  """
+
+
+  @classmethod
+  def getSpec(cls):
+    ns = dict(
+      description=SDRClassifierRegion.__doc__,
+      singleNodeOnly=True,
+
+      inputs=dict(
+        categoryIn=dict(
+          description='Vector of categories of the input sample',
+          dataType='Real32',
+          count=0,
+          required=True,
+          regionLevel=True,
+          isDefaultInput=False,
+          requireSplitterMap=False),
+
+        bottomUpIn=dict(
+          description='Belief values over children\'s groups',
+          dataType='Real32',
+          count=0,
+          required=True,
+          regionLevel=False,
+          isDefaultInput=True,
+          requireSplitterMap=False),
+
+        predictedActiveCells=dict(
+          description="The cells that are active and predicted",
+          dataType='Real32',
+          count=0,
+          required=True,
+          regionLevel=True,
+          isDefaultInput=False,
+          requireSplitterMap=False),
+
+        sequenceIdIn=dict(
+          description="Sequence ID",
+          dataType='UInt64',
+          count=1,
+          required=False,
+          regionLevel=True,
+          isDefaultInput=False,
+          requireSplitterMap=False),
+      ),
+
+      outputs=dict(
+        categoriesOut=dict(
+          description='Classification results',
+          dataType='Real32',
+          count=0,
+          regionLevel=True,
+          isDefaultOutput=False,
+          requireSplitterMap=False),
+
+        actualValues=dict(
+          description='Classification results',
+          dataType='Real32',
+          count=0,
+          regionLevel=True,
+          isDefaultOutput=False,
+          requireSplitterMap=False),
+
+        probabilities=dict(
+          description='Classification results',
+          dataType='Real32',
+          count=0,
+          regionLevel=True,
+          isDefaultOutput=False,
+          requireSplitterMap=False),
+      ),
+
+      parameters=dict(
+        learningMode=dict(
+          description='Boolean (0/1) indicating whether or not a region '
+                      'is in learning mode.',
+          dataType='UInt32',
+          count=1,
+          constraints='bool',
+          defaultValue=1,
+          accessMode='ReadWrite'),
+
+        inferenceMode=dict(
+          description='Boolean (0/1) indicating whether or not a region '
+                      'is in inference mode.',
+          dataType='UInt32',
+          count=1,
+          constraints='bool',
+          defaultValue=0,
+          accessMode='ReadWrite'),
+
+        maxCategoryCount=dict(
+          description='The maximal number of categories the '
+                      'classifier will distinguish between.',
+          dataType='UInt32',
+          required=True,
+          count=1,
+          constraints='',
+          # arbitrarily large value for backward compatibility 
+          defaultValue=1000,
```

Comment:
It seems that this just needs to be a large enough value. It is arbitrarily set to be 1000